--- zconf.h     Wed Jul 20 17:40:26 2005
+++ zconf.h     Sun May 21 14:38:10 2006
@@ -284,7 +284,7 @@
    typedef Byte       *voidp;
 #endif
 
-#if 1           /* HAVE_UNISTD_H -- this line is updated by ./configure */
+#if 0           /* HAVE_UNISTD_H -- this line is updated by ./configure */
 #  include <sys/types.h> /* for off_t */
 #  include <unistd.h>    /* for SEEK_* and off_t */
 #  ifdef VMS
